ra/ra.go

```diff
@@ -1427,13 +1427,20 @@ func (ra *RegistrationAuthorityImpl) checkCertificatesPerNameLimit(ctx context.C
 				return nil
 			}
 		}
-		domains := strings.Join(namesOutOfLimit, ", ")
+
+		ra.log.Infof("Rate limit exceeded, CertificatesForDomain, regID: %d, domains: %s", regID, strings.Join(namesOutOfLimit, ", "))
 		ra.certsForDomainStats.Inc("Exceeded", 1)
-		ra.log.Infof("Rate limit exceeded, CertificatesForDomain, regID: %d, domains: %s", regID, domains)
-		return berrors.RateLimitError(
-			"too many certificates already issued for: %s",
-			domains,
-		)
+		if len(namesOutOfLimit) > 1 {
+			var subErrors []berrors.SubBoulderError
+			for _, name := range namesOutOfLimit {
+				subErrors = append(subErrors, berrors.SubBoulderError{
+					Identifier:   identifier.DNSIdentifier(name),
+					BoulderError: berrors.RateLimitError("too many certificates already issued").(*berrors.BoulderError),
+				})
+			}
+			return berrors.RateLimitError("too many certificates already issued for multiple names").(*berrors.BoulderError).WithSubErrors(subErrors)
```

Comment:
I think the top level problem needs to have at least one of the `namesOutOfLimit` or this will be a bit of a regression information-wise for the majority of clients that haven't implemented displaying subproblem information when errors occur.